### PR TITLE
Added API handler for fetching observation dates given a list of entities and variables

### PR DESCRIPTION
**Pull request by dwnoble**
Added API handler for fetching observation dates given a list of entities and variables.

New website API endpoint: `/api/observation-dates/entities`. This endpoint is similar to the `/api/observation-dates` endpoint, but it takes a list of entities instead of a parentEntity and childType.

This endpoint will support updating the timeline slider web component to accept a list of entities rather than parent/child relationships.

Example usage:
  ```
  GET /api/observation-dates/entities?entities=country/USA&entities=country/CAN&variables=Count_Person&variables=Count_Household
  Response:
  {
      "datesByVariable": [
          {
              "variable": "Count_Person",
              "observationDates": [
                  {
                      "date": "1900",
                      "entityCount": [
                          {
                              "facet": "2176550201",
                              "count": 1
                          }
                      ]
                  },
                  {
                      "date": "1901",
                      "entityCount": [
                          {
                              "facet": "2176550201",
                              "count": 1
                          }
                      ]
                  }
              ]
          },
          {
              "variable": "Count_Household",
              "observationDates": [
                  {
                      "date": "1900",
                      "entityCount": [
                          {
                              "facet": "2176550202",
                              "count": 1
                          }
                      ]
                  },
                  {
                      "date": "1901",
                      "entityCount": [
                          {
                              "facet": "2176550202",
                              "count": 1
                          }
                      ]
                  }
              ]
          }
      ],
      "facets": {
          "2176550201": {"importName": "facet1"},
          "2176550202": {"importName": "facet2"}
      }
  }
  ```

### CODE REVIEW

**Review by chejennifer — COMMENTED** [review]
nice! just small nits

**Review by juliawu — COMMENTED** [review]
Thank you for adding tests! This lgtm, deferring to @chejennifer for approval since she has more comments.

File: server/lib/util.py [hunk 1]
```diff
@@ -527,6 +529,232 @@ def gzip_compress_response(raw_content, is_json):
...
+      observations_for_date = list(observations_for_date_group)
+      observations_for_date.sort(key=itemgetter('facet'))
+      for facet_key, observations_for_facet_group in groupby(
+          observations_for_date, key=itemgetter('facet')):
+        entity_count_item = {
+            'count': len(list(observations_for_facet_group)),
+            'facet': facet_key
+        }
+        observation_dates_item['entityCount'].append(entity_count_item)
+  return dates_by_variable
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply server/lib/util.py:652]: Done!

File: server/lib/util.py [hunk 2]
```diff
@@ -527,6 +529,232 @@ def gzip_compress_response(raw_content, is_json):
...
+      {'date': '1900', 'entity': 'country/USA', 'facet': '2176550201', 'value': 76094000, 'variable': 'Count_Person'},
+      {'date': '1901', 'entity': 'country/USA', 'facet': '2176550201', 'value': 77584000, 'variable': 'Count_Person'}
+  ]
+  """
+  flattened_observations = []
+  variables = obs_series_response["byVariable"].keys()
+  for variable in variables:
+    entities = obs_series_response["byVariable"][variable]["byEntity"].keys()
+    for entity in entities:
+      variable_entity_entry = obs_series_response["byVariable"][variable][
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply server/lib/util.py:572]: Done

File: server/lib/util.py [hunk 3]
```diff
@@ -527,6 +529,232 @@ def gzip_compress_response(raw_content, is_json):
...
+          "byEntity"][entity]
+      if not variable_entity_entry:
+        continue
+      for ordered_facet in variable_entity_entry['orderedFacets']:
+        for observation in ordered_facet['observations']:
+          flattened_observations.append({
+              'date': observation['date'],
```
[the remark this answers is not in the document]
Comment (dwnoble) [reply server/lib/util.py:579]: Added getter for 'value' since it's technically optional

**Review by chejennifer — APPROVED** [review]
thanks for updating!

**Pull request merged**